qiskit/quantum_info/primitives/sampler/appendable_sampler.py

```diff
@@ -0,0 +1,82 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Appendable sampler
+"""
+
+from __future__ import annotations
+
+import logging
+from typing import Optional, Union
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.providers.backend import BackendV1 as Backend
+from qiskit.result import Result
+
+from ..backends import BackendWrapper
+from ..results import BaseResult, SamplerResult
+from .sampler import Sampler
+
+logger = logging.getLogger(__name__)
+
+
+class AppendableSampler(Sampler):
+    """Sampler class that can append results"""
+
+    def __init__(
+        self,
+        backend: Union[Backend, BackendWrapper],
+        circuits: Optional[Union[QuantumCircuit, list[QuantumCircuit]]] = None,
+    ):
+        super().__init__(backend=backend, circuits=circuits)
+        self._raw_results: list[Result] = []
+
+    # pylint: disable=arguments-differ
```

Comment:
Same as estimator. Do we need to define run at the `BasePrimitive` level if it's child classes all override it independently?